toqito/state_props/is_abs_ppt.py

```diff
@@ -0,0 +1,121 @@
+
+    Only implemented for 2x2, 2x3, and 3x3 systems.
+    """
+    eigvals = np.sort(np.real(eigvals))[::-1]
+    d1, d2 = dim
+    constraints = []
+
+    if (d1, d2) == (2, 2):
+        # Only one constraint: 2x2 matrix
+        # See Hildebrand, Eq. (2)
+        # Matrix: [[2*lam4, lam3-lam1], [lam3-lam1, 2*lam2]]
+        L = np.array(
+            [
+                [2 * eigvals[3], eigvals[2] - eigvals[0]],
+                [eigvals[2] - eigvals[0], 2 * eigvals[1]],
+            ]
+        )
+        constraints.append(L)
+    elif (d1, d2) in [(2, 3), (3, 2)]:
+        # Two constraints: 3x3 matrices
+        # See Hildebrand, Eq. (3)
+        # eigvals: lam1 >= ... >= lam6
+        L1 = np.array(
+            [
+                [2 * eigvals[5], eigvals[4] - eigvals[0], eigvals[3] - eigvals[1]],
+                [eigvals[4] - eigvals[0], 2 * eigvals[2], eigvals[1] - eigvals[3]],
+                [eigvals[3] - eigvals[1], eigvals[1] - eigvals[3], 2 * eigvals[0]],
+            ]
+        )
+        L2 = np.array(
+            [
+                [2 * eigvals[5], eigvals[4] - eigvals[0], eigvals[2] - eigvals[1]],
+                [eigvals[4] - eigvals[0], 2 * eigvals[3], eigvals[1] - eigvals[2]],
+                [eigvals[2] - eigvals[1], eigvals[1] - eigvals[2], 2 * eigvals[0]],
+            ]
+        )
+        constraints.extend([L1, L2])
+    elif (d1, d2) == (3, 3):
+        # Two constraints: 3x3 matrices
+        # See QETLAB/AbsPPTConstraints and Hildebrand, Eq. (4)
+        # eigvals: lam1 >= ... >= lam9
+        L1 = np.array(
+            [
+                [2 * eigvals[8], eigvals[7] - eigvals[0], eigvals[6] - eigvals[1]],
+                [eigvals[7] - eigvals[0], 2 * eigvals[5], eigvals[4] - eigvals[2]],
+                [eigvals[6] - eigvals[1], eigvals[4] - eigvals[2], 2 * eigvals[3]],
+            ]
+        )
+        L2 = np.array(
+            [
+                [2 * eigvals[8], eigvals[7] - eigvals[0], eigvals[5] - eigvals[1]],
+                [eigvals[7] - eigvals[0], 2 * eigvals[6], eigvals[4] - eigvals[2]],
+                [eigvals[5] - eigvals[1], eigvals[4] - eigvals[2], 2 * eigvals[3]],
+            ]
+        )
+        constraints.extend([L1, L2])
+    else:
+        # Not implemented for higher dimensions
+        raise NotImplementedError("Absolutely PPT constraints for dimensions > 3x3 are not implemented.")
+    return constraints
+
+
+def is_abs_ppt(mat: np.ndarray, dim: None | int | list[int] = None, tol: float = 1e-8) -> bool | int:
+    """Determine whether a density matrix is absolutely PPT (PPT from spectrum).
+
+    Args:
+        mat: The density matrix to check.
+        dim: The local dimensions (default: inferred as equal dims).
+        tol: Numerical tolerance for positive semidefinite checks.
+
+    Returns:
+        True if absolutely PPT, False if not, -1 if undecidable (for large dims).
+
```

Comment:
@luffy-orf We have a different style for the docstrings. Could you take a look at the contributing guide in the documentation?